blocks/log/json.py

```diff
@@ -0,0 +1,83 @@
+from mimir import Logger
+from mimir.logger import _Logger
+
+from .log import TrainingLogBase
+
+
+class PicklableLogger(_Logger):
+    """A picklable wrapper around mimir logger.
+
+    This class is a picklable version of `:class:mimir.Logger`.
+
+    """
+    def __init__(self, **kwargs):
+        self.logger_kwargs = kwargs
+        self.opened = False
+
+    def open(self):
+        if not self.opened:
+            logger = Logger(**self.logger_kwargs)
+            self.__dict__.update(logger.__dict__)
+            self.load(self.logger_kwargs['filename'])
+            self.opened = True
+
+    def __setstate__(self, state):
+        self.logger_kwargs = state
+        self.opened = False
+        self.open()
+
+    def __getstate__(self):
+        return self.logger_kwargs
+
+
+class JSONLinesLog(TrainingLogBase):
+    """JSON Lines log.
+
+    The current status is saved in
+    `iteration_status` attribute and is flushed to the file only if
+    the next iteration is requested.
+
+    """
+    def __init__(self, filename='log.jsonl.gz', **kwargs):
+        self.status = {}
+        TrainingLogBase.__init__(self)
+        kwargs.setdefault("maxlen", 2)
+        kwargs.setdefault("filename", filename)
+        kwargs.setdefault("formatter", None)
+        self.logger = PicklableLogger(**kwargs)
+        self.last_flushed = -1
+        self.iteration_status = {}
```

Comment:
Does it contain the last row of the log? If yes, why not call it `current_row`? Status is a name reserved specifically for status, I thought.